wit/store.wit

```diff
@@ -0,0 +1,71 @@
+/// A keyvalue interface that provides eventually consistent key-value operations.
+/// 
+/// Each of these operations acts on a single key-value pair.
+/// 
+/// The value in the key-value pair is defined as a `u8` byte array and the intention is that it is
+/// the common denominator for all data types defined by different key-value stores to handle data,
+/// ensuring compatibility between different key-value stores. Note: the clients will be expecting
+/// serialization/deserialization overhead to be handled by the key-value store. The value could be
+/// a serialized object from JSON, HTML or vendor-specific data types like AWS S3 objects.
+/// 
+/// Data consistency in a key value store refers to the guarantee that once a write operation
+/// completes, all subsequent read operations will return the value that was written.
+/// 
+/// Any implementation of this interface must have enough consistency to guarantee "reading your
+/// writes." In particular, this means that you should never get a value that is older than the one
+/// you wrote, but you MAY get a newer value if one was written around the same time. For example,
+/// if a value at a given key is A, and you write B, then immediately read, you should get B. If
+/// something else writes C in quick succession, then you may get C.
+interface store {
+    use types.{bucket, error};
+
+    /// The crud resource contains the common create, read, update, and delete operations for
+    /// key-value pairs in a key-value store
+    resource store {
+        /// Create a new crud resource, using the given bucket to access the data
+        open: static func(store: borrow<bucket>) -> store;
+
+        /// Get the value associated with the specified `key`
+        ///
+        /// The value is returned as an option. If the key-value pair exists in the
+        /// store, it returns `Ok(value)`. If the key does not exist in the
+        /// store, it returns `Ok(none)`. 
+        ///
+        /// If any other error occurs, it returns an `Err(error)`.
+        get: func(key: string) -> result<option<list<u8>>, error>;
+
+        /// Set the value associated with the key in the store. If the key already
+        /// exists in the store, it overwrites the value.
+        ///
+        /// If the key does not exist in the store, it creates a new key-value pair.
+        /// 
+        /// If any other error occurs, it returns an `Err(error)`.
+        set: func(key: string, value: list<u8>) -> result<_, error>;
```

Comment:
There is a concept of ["SessionToken"](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.documents.client.requestoptions.sessiontoken?view=azure-dotnet#remarks) for Cosmos DB if you set the consistency level to session consistency (which guarantees read-your-write). This token is used with each read request to ensure that the client receives the most recent commited version of a kv pair within the session. 

 I would like to know your thoughts on this!

Reply:
So I have definitely used tokens like this before, but in practice, I think it would make the API much more complicated, especially since other key value APIs don't have this by default (they often have it somewhere, but not at their "top level" API as it were). I am shooting for ease of use based on what we've seen the users of our projects use as well as in [other](https://www.fastly.com/documentation/reference/api/services/resources/kv-store-item/) [APIs](https://developers.cloudflare.com/kv/api/) [out](https://redis.io/commands/get/) [there](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GettingStarted.ReadItem.html). (That last example from Dynamo is telling because they also allow you to select a consistency model, but the example shows just fetching by name)